proxy/src/control/destination/background.rs

```diff
@@ -571,6 +607,67 @@ impl<T: HttpService<ResponseBody = RecvBody>> DestinationSet<T> {
     }
 }
 
+// ===== impl BindClient =====
+
+impl BindClient {
+    fn new(
+        identity: Conditional<tls::Identity, tls::ReasonForNoTls>,
+        dns_resolver: &dns::Resolver,
+        host_and_port: HostAndPort,
+    ) -> Self {
+        let scheme = http::uri::Scheme::from_shared(Bytes::from_static(b"http")).unwrap();
+        let authority = http::uri::Authority::from(&host_and_port);
+        let log_ctx = ::logging::admin().client("control", host_and_port.clone());
+        Self {
+            identity,
+            dns_resolver: dns_resolver.clone(),
+            host_and_port,
+            authority,
+            scheme,
+            log_ctx,
+        }
+    }
+}
+impl Rebind<tls::ConditionalClientConfig> for BindClient {
+    type Service = ClientService;
+    fn rebind(
+        &mut self,
+        client_cfg: &tls::ConditionalClientConfig,
+    ) -> Self::Service {
+        // Join the `Conditional<tls::Identity>` in `self.identity` with the
+        // passed-in `Conditional<tls::ClientConfig>` to build a
+        // `tls::ConnectionConfig` (or `None` if either is `None).
```

Comment:
This seems a bit overly-explanatory, especially if you take my suggestion of using a match statement.